lib/pcsclite.js

```diff
@@ -90,6 +90,10 @@ CardReader.prototype.connect = function(options, cb) {
     options.share_mode = options.share_mode || this.SCARD_SHARE_EXCLUSIVE;
     options.protocol = options.protocol || this.SCARD_PROTOCOL_T0 | this.SCARD_PROTOCOL_T1;
 
+    if (options.share_mode = this.SCARD_SHARE_DIRECT) {
+      options.protocol =  0;
+    }
```

Comment:
I think would prefer allowing `option.protocol` to be 0. If the user passes an invalid value, then I guess connect should fail. What do you think?